src/app/lib/utilities/passport/index.js

```diff
@@ -28,3 +28,19 @@ export const isValidPassportHome = (
 
   return passportHomesOverrideArray.some(matchesPassportHome);
 };
+
+export const getCanonicalUrl = pageData => {
+  const canonicalUrl = pathOr(
+    null,
+    ['metadata', 'locators', 'canonicalUrl'],
+    pageData,
+  );
+  return canonicalUrl
+    ? canonicalUrl.replace(/.*(bbc\.com|bbc\.co\.uk|localhost:\d*)/, '')
+    : null;
+};
+
+export const matchesCanonicalUrl = (canonicalUrl, pathName) => {
```

Comment:
Just for my understanding and because I'm reviewing this on my phone, what situations would the path not match the canonical URL?

Reply:
Heya, so we added this logic to ensure that Sports articles with a discipline will only render at the correct URL.

At present, an asset intended to be rendered at /sport/football/articles/:id will also render at /sport/articles/:id. We only want it to render at /sport/football/articles/:id

The canonicalUrl from the BFF response includes the discipline. So if the pathname is missing the discipline, then the article will not render.

(I think that's correct at least! I might have got muddled on what this exact bit of logic does)